python/dlisio/plumbing/errorhandler.py

```diff
@@ -0,0 +1,216 @@
+             developer's assumption. Developers haven't seen these errors in the
+             real world and don't know if recovery is ok. Usually would appear
+             when accessing attributes.
+
+             By default a warning is logged.
+    minor    Any issue that dlisio is able to recover from under light
+             developer's assumption. Developers have seen this error in real
+             files and generally believe that data is presented to the users
+             correctly.
+
+             May also appear when nearby data is spoiled, hence hinting user
+             to be cautious.
+
+             By default an info message is logged.
+    info     Issue doesn't contradict specification, but situation is peculiar.
+
+             By default a debug message is logged.
+
+    ======== ===================================================================
+
+    Error handler does not cover all the warnings and exceptions.
+    It is mostly intended to be used for data parsing, not for data
+    presentation. Thus it would be applicable, for example
+
+    * on data load and parsing in :py:func:`dlisio.load()`,
+      :py:func:`dlisio.logicalfile.load()` and
+      :py:func:`dlisio.plumbing.Frame.curves()`
+    * on object access :py:func:`dlisio.logicalfile.find()`,
+      :py:func:`dlisio.logicalfile.object()`
+    * on attribute access :py:attr:`dlisio.plumbing.Frame.description`
+
+    User still gets default behavior on other occasions, like warnings on
+    missing SUL or exceptions in :py:func:`dlisio.plumbing.Frame.curves()` in
+    case of invalid channel/frame relationships. Some of these might be get
+    implemented in the future.
+
+    Please also note that error handle doesn't redefine issues categories, it
+    only changes default behavior.
+
+    Attributes
+    ----------
+    info:     :py:class:`ErrorHandler.Actions`
+        Action for merely information message
+
+    minor:    :py:class:`ErrorHandler.Actions`
+        Action for minor specification violation
+
+    major:    :py:class:`ErrorHandler.Actions`
+        Action for major specification violation
+
+    critical: :py:class:`ErrorHandler.Actions`
+        Action for critical specification violation
+
+
+    Warnings
+    --------
+    Escaping errors is a good solution when user needs to read as much data as
+    possible, for example, to have a general overview over the file. However
+    user must be careful when using this mode during close inspection.
+    If user decides to accept errors, they must be aware that some returned data
+    will be spoiled. Most likely it will be data which is stored in the file
+    near the failure.
+
+    Warnings
+    --------
+    Be careful not to ignore too much information when investigating files.
+    dlisio might classify some issue as a minor violation, but it might be
+    critical for understanding the true level of data brokeness.
```

Comment:
What is "the true level of data brokenness", and when do I need to understand it?

I think what you are essentially trying to say is that "if you want to debug a broken part of the file, you should look at _all_ errors and warnings (from that part) to get a full picture of the situation." Or am I mistaken?